### PR TITLE
Method chaining indentation fixer

### PR DESCRIPTION
**Pull request by julienfalque**
Replaces #2617 with some refactoring and small fixes.

/cc @boliev

### CODE REVIEW

**Comment by fterrag** [comment]
@julienfalque thanks for opening this PR. Any updates here?

**Comment by julienfalque** [comment]
This PR mainly needs proper review for now :)

**Comment by fterrag** [comment]
@julienfalque alright, cool! Who can we ping to get some movement here?

File: src/Fixer/Whitespace/MethodChainingIndentationFixer.php [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+ * @author Vladimir Boliev <voff.web@gmail.com>
+ */
+final class MethodChainingIndentationFixer extends AbstractFixer implements WhitespacesAwareFixerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
```
Comment (keradus) [inline src/Fixer/Whitespace/MethodChainingIndentationFixer.php]: a note that it is not supposed to work for tree-indented calls (eg like with Sf compilers) would make sense

Comment (julienfalque) [reply src/Fixer/Whitespace/MethodChainingIndentationFixer.php]: Done.

**Comment by julienfalque** [comment]
@fterrag If you want this PR to be merged faster (and if you have time to), please help by reviewing :)

**Comment by kubawerlos** [comment]
What about indenting to the arrow? Like this:
```php
$somethingBuilder->addDescription('Hello')
                 ->addNumber(42)
                 ->addSomethingElse('something else');
```
Doctrine does that in their [docs](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/query-builder.html#binding-parameters-to-your-query).

Also an option if the first chaining should be in the line with the variable or in the next one would be useful.

**Comment by julienfalque** [comment]
Why not but I'd like to avoid implementing too much things here and merge the current PR first.

**Comment by keradus** [comment]
btw, please incorporate my commits into @boliev 's ones next time you will make a rebase

**Comment by keradus** [comment]
please make a rebase @julienfalque

**Comment by julienfalque** [comment]
Rebased. I'm not sure the CircleCI failure is related to changes in this PR as I'm getting similar errors locally with master.

**Comment by keradus** [comment]
Thank you @julienfalque.

**Comment by keradus** [comment]
Thank you @boliev.

**Comment by cviebrock** [comment]
Just wondering if anything became of @kubawerlos 's [comment](#issuecomment-345355625) for allowing aligning chained methods on the arrow, e.g.:
```php
$somethingBuilder->addDescription('Hello')
                 ->addNumber(42)
                 ->addSomethingElse('something else');
```
This is still not possible in 2.12.2, correct?

**Comment by SpacePossum** [comment]
I don't think this has been done, there isn't an RFC for it as well. If you want you create an issue describing the RFC or maybe make a PR :)

**Comment by TomasVotruba** [comment]
Hi guys, I use this rule to indent and it works well.
But that is not enough, since fluent call create->super()->long()->lines()->that()->shouldBe()->readable()

So I made a fixer that breaks fluent lines to per-call line:

https://github.com/symplify/coding-standard/blob/0bd55495794d3b5380795a2e4c551267641fbab7/src/Fixer/Spacing/MethodChainingNewlineFixer.php

Feel free to use it, copy it, improve it #foss :+1: 

```diff
 class SkipMultiLineChain
 {
     public function run()
     {
-        return $this->one()->two();
+        return $this->one()
+->two();
     }

 }
```

**Pull request merged**